Assets/BowAndArrowImproved/Scripts/TeleportManager.cs

```diff
@@ -0,0 +1,61 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TeleportManager : MonoBehaviour
+{
+    [SerializeField] private GameObject playerContainer;
+    [SerializeField] private List<TeleportAnchor> anchorsList;
+
+    private Coroutine _teleportCoroutine;
+    
+    private void OnEnable()
+    {
+        _teleportCoroutine = StartCoroutine(TeleportCoroutine());
+    }
+
+    private void OnDisable()
+    {
+        if (_teleportCoroutine != null)
+        {
+            StopCoroutine(_teleportCoroutine);
+        }
+    }
+
+    private IEnumerator TeleportCoroutine()
```

Comment:
why is this called TeleportCoroutine? nothing about this coroutine teleports anything, it simple enables/disabled the anchors, please change the name